include/mpsl/mpsl_clock_ctrl_cb.h

```
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @file mpsl_clock_ctrl_cb.h
+ *
+ * @defgroup mpsl_clock_ctrl_cb MPSL Clock Control Notification Callback API
+ *
+ * @brief Callback API for MPSL clock control event notifications.
+ * @{
+ */
+
+#ifndef MPSL_CLOCK_CTRL_CB_H__
+#define MPSL_CLOCK_CTRL_CB_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ *  @typedef mpsl_clock_ctrl_cb_t
+ *
+ *  @brief Callback type for clock control event.
+ */
+typedef void (*mpsl_clock_ctrl_cb_t)(void);
+
+/**
+ * @brief Register callback for HFCLK requested event.
+ *
+ * @param cb Function to call after HFCLK is requested by MPSL. @c NULL to unregister.
+ */
+void mpsl_clock_ctrl_cb_register_hfclk_requested(mpsl_clock_ctrl_cb_t cb);
```

I think the register callback functions either needs to handle multiple callbacks being registered (e.g. like gpio callbacks are registered: https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/gpio/gpio_utils.h#L59), or return an error when a callback is already set. With this design you will get silent errors when there are multiple users.

If you opt for a list of callbacks, it would probably be easier to have a single callback with an extra argument that indicates if HFCLK is requested or released instead of having separate callbacks.

I would also recommend writing a `@note` or similar in the function description to make it clear that the callback can potentially be called from a higher priority thread. I don't know MPSL well enough to understand all paths from where the HFCLK requests/releases can be made